subsys/bluetooth/host/l2cap.c

```diff
@@ -31,6 +31,8 @@
 
 #define L2CAP_LE_MIN_MTU		23
 #define L2CAP_ECRED_MIN_MTU		64
+#define L2CAP_ECRED_MAX_MTU		MIN(CONFIG_BT_L2CAP_TX_MTU, \
```

Comment:
The thing with these L2CAP tests is that this EATT SPSM is used only as standard PSM for tests, but doesn't really test nor requires EATT. These tests can be executed with EATT disabled, and just BT_L2CAP_ECRED is required. In fact, this is how I test it. We just need to register server with EATT PSM and listen on it. There is test spec errata filed and accepted (TSE 16780), but TCLR is not yet released. There should be IXIT in PTS to set what PSM shall be used.

Second of all, yes, L2CAP/ECFC/BV-27-C was the reason for checking max MTU. I should've added this in commit message. I read through TS some more and I think there are multiple issues with this test: first, it requires to set this max MTU value in IXIT, but no such IXIT exists (only for ATT MTU). Secondly, the test passes even if no error is sent and L2CAP connection on EATT PSM is successful. What's more, because no value is set in IXIT, MTU in request is set to 0xFFFF, which should end test as inconclusive. I think this requires PTS errata.

To sum it up, I'll just remove this max MTU check and just keep  BT_L2CAP_LE_ERR_INVALID_PARAMS error.
I'll also file an PTS support case for L2CAP/ECFC/BV-27-C

Reply:
Ok, thank you for the explanation. I agree with you that it is best to have this resolved with PTS first before we make a change here.
The thing that I couldn't understand from the test is why we would say the MTU is too large, since the MTU the remote is sending is the MTU that it can receive, it shouldn't be a problem that it is larger than the packets that we are sending.